beetsplug/fetchart.py

```diff
@@ -230,22 +231,42 @@ def fetch_image(self, candidate, extra):
             with closing(self.request(candidate.url, stream=True,
                                       message=u'downloading image')) as resp:
                 ct = resp.headers.get('Content-Type', None)
-                if ct not in CONTENT_TYPES:
-                    self._log.debug(
-                        u'not a supported image: {}',
-                        resp.headers.get('Content-Type') or u'no content type',
-                    )
-                    candidate.path = None
-                    return
-
-                # Generate a temporary file with the correct extension.
-                with NamedTemporaryFile(suffix=b'.' + CONTENT_TYPES[ct][0],
-                                        delete=False) as fh:
-                    for chunk in resp.iter_content(chunk_size=1024):
+
+                # Generate a temporary file and guess the extension based on
+                # the Content-Type header. This may be wrong for badly
+                # configured servers. E.g. fanart.tv only allows .jpg uploads
+                # and ALWAYS returns a image/jpeg Content-Type, but other
+                # formats with a erroneous .jp(e)g extension apparently can
+                # sneak through the upload filter. Therefore validate the type
+                # using the file magic.
+                data = resp.iter_content(chunk_size=1024)
```

Comment:
@jackwilsdon, your test still gets the whole file and under different circumstances as fetchart uses the `stream=True` option. For valid images, this would contain all of the header. The interesting part (to my understanding, and I really don't know anything about networking) would be what happens when during streaming the server sends packets smaller than `chunk_size`.

However, in any case my code is a regression when for some reason no more than a few bytes are returned at all (corrupt image, maybe connectivity issues?), so a loop would be the way to go. Else, the whole thing could crash with an `IndexError` or the like in `_image_mime_type`.

And I think a loop is cleaner anyway than the `try`...`except` I got right now.